src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -99,16 +99,19 @@ private function addRoute(RouteCollection $routeCollection, string $resourceClas
             throw new RuntimeException('Either a "route_name" or a "method" operation attribute must exist.');
         }
 
-        if (isset($operation['controller'])) {
-            $controller = $operation['controller'];
-        } else {
-            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collection ? 'collection' : 'item');
-            $controller = self::DEFAULT_ACTION_PATTERN.$actionName;
-        }
+        $collectionString = $collection ? 'collection' : 'item';
+
+        $actionName = sprintf('%s_%s', $operationName, $collectionString);
+        $controller = $operation['controller'] ?? self::DEFAULT_ACTION_PATTERN.sprintf('%s_%s', strtolower($operation['method']), $collectionString);
```

Comment:
Indeed, but if no controller is specified it's better to keep the default
one, less error prone, don't you agree?
Also, this is the same behavior as in 1.x
Le ven. 8 avr. 2016 à 13:14, Teoh Han Hui <notifications@github.com> a
écrit :

> In src/Bridge/Symfony/Routing/ApiLoader.php
> <https://github.com/api-platform/core/pull/498#discussion_r59010578>:
>
> > @@ -99,16 +99,19 @@ private function addRoute(RouteCollection $routeCollection, string $resourceClas
> >              throw new RuntimeException('Either a "route_name" or a "method" operation attribute must exist.');
> >          }
> >
> > -        if (isset($operation['controller'])) {
> > -            $controller = $operation['controller'];
> > -        } else {
> > -            $actionName = sprintf('%s_%s', strtolower($operation['method']), $collection ? 'collection' : 'item');
> > -            $controller = self::DEFAULT_ACTION_PATTERN.$actionName;
> > -        }
> > +        $collectionString = $collection ? 'collection' : 'item';
> > +
> > +        $actionName = sprintf('%s_%s', $operationName, $collectionString);
> > +        $controller = $operation['controller'] ?? self::DEFAULT_ACTION_PATTERN.sprintf('%s_%s', strtolower($operation['method']), $collectionString);
>
> With my code:
>
> $actionName = sprintf('%s_%s', strtolower($operationName), $collection ? 'collection' : 'item');$controller = $operation['controller'] ?? self::DEFAULT_ACTION_PATTERN.$actionName;
>
> This is supported:
>
> /** * @Resource(itemOperations={ *     "get"={"method"="GET"}, * }) */
>
> while this is not:
>
> /** * @Resource(itemOperations={ *     "my_get_op"={"method"="GET"}, * }) */
>
> but you could specify it explicitly:
>
> /** * @Resource(itemOperations={ *     "my_get_op"={"method"="GET", "controller"="api_platform.action.get_item"}, * }) */
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/api-platform/core/pull/498/files/bed469a35ec4102c8a9c456fb2d7ab34d8b75d6a#r59010578>
>